src/site/cpp-arrays.rst

```diff
@@ -158,6 +158,107 @@ Arrays vs. Pointers
 Dynamic Arrays
 ==============
 
+* The static arrays require a known size at compile time
+* This is because cpp wants to allocate the RAM on the stack it needs before it starts running
+    * Cpp will manage the memory on the stack for us
+
+* This means determining the size of the array at runtime is problematic
+* The below example will not work
+    * On some platforms this will actually work, but assume it will not
+
+.. code-block:: cpp
+    :linenos:
+
+    int size;
+    std::cin >> size;
+    int foo[size];
+
+
+* However, it is not unreasonable to want to have an array that's size is determined at runtime
+* Fortunately, there is still a way to do this, but we need to allocate the memory on the *heap*
+    * Cpp does not manage this memory for us
+    * We must take care to allocate and deallocate the memory we use in the heap
```

Comment:
Worth mentioning failing to free memory leads to a memory leak. I imagine they'll be familiar with the term but not actually know what it refers to.